From 936c857add300f41bc58c300793a0e10b48ff69f Mon Sep 17 00:00:00 2001
From: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>
Date: Mon, 13 Dec 2021 15:25:23 +0000
Subject: [PATCH 19/24] Use correct buffer size

The comm buffer created has additional 4 bytes length which
needs to be trimmed. This change will reduce the size of the
comm buffer to what is expected.

Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 include/mm_communication.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/mm_communication.h b/include/mm_communication.h
index e65fbde60d0a..bb9919095649 100644
--- a/include/mm_communication.h
+++ b/include/mm_communication.h
@@ -123,7 +123,7 @@ struct __packed efi_mm_communicate_header {
  *
  * Defined in EDK2 as SMM_VARIABLE_COMMUNICATE_HEADER.
  */
-struct smm_variable_communicate_header {
+struct __packed smm_variable_communicate_header {
 	efi_uintn_t  function;
 	efi_status_t ret_status;
 	u8           data[];
@@ -145,7 +145,7 @@ struct smm_variable_communicate_header {
  * Defined in EDK2 as SMM_VARIABLE_COMMUNICATE_ACCESS_VARIABLE.
  *
  */
-struct smm_variable_access {
+struct __packed smm_variable_access {
 	efi_guid_t  guid;
 	efi_uintn_t data_size;
 	efi_uintn_t name_size;
-- 
2.37.1

